docs/open-api-v3-support.md

```diff
@@ -0,0 +1,320 @@
+
+## General
+
+### [Data Types](https://github.com/OAI/OpenAPI-Specification/blob/OpenAPI.next/versions/3.0.md#dataTypes)
+
+- [x] Primitive data types, `integer`, `number`, `string` and `boolean`.
+- [x] Any `format` will be displayed.
+
+### [Rich Text Formatting](https://github.com/OAI/OpenAPI-Specification/blob/OpenAPI.next/versions/3.0.md#rich-text-formatting)
+
+- [x] All `description` fields support [CommonMark][cm].
+
+### [Relative References in URLs](https://github.com/OAI/OpenAPI-Specification/blob/OpenAPI.next/versions/3.0.md#relative-references-in-urls)
+
+- [ ] Relative references are resolved using the URLs defined in the Server Object as a Base URI.
+- [x] Relative references used in `$ref` are processed as per JSON Reference.
+
+## Schema
+
+### [OpenAPI](https://github.com/OAI/OpenAPI-Specification/blob/OpenAPI.next/versions/3.0.md#openapi-object) object
+
+- [ ] openapi
+- [x] [info](#info-object)
+- [ ] [servers](#servers-object)
+- [x] [paths](#paths-object)
+- [ ] [components](#components-object)
+- [ ] [security](#security-requirement-object)
+- [ ] [tags](#tag-object)
+- [ ] [externalDocs](#external-documentation-object)
+
+### [Info](https://github.com/OAI/OpenAPI-Specification/blob/OpenAPI.next/versions/3.0.md#info-object) object
+
+- [x] title
+- [x] description
+- [ ] termsOfService
+- [ ] [contact](#contact-object)
+- [ ] [license](#license-object)
+- [x] version
+
+### [Contact](https://github.com/OAI/OpenAPI-Specification/blob/OpenAPI.next/versions/3.0.md#contact-object)
+
+- [ ] name
+- [ ] url
+- [ ] email
+
+### [License](https://github.com/OAI/OpenAPI-Specification/blob/OpenAPI.next/versions/3.0.md#license-object) object
+
+- [ ] name
+- [ ] url
+
+### [Server](https://github.com/OAI/OpenAPI-Specification/blob/OpenAPI.next/versions/3.0.md#server-object) object
+
+- [ ] url
+- [ ] description
+- [ ] [variables](#server-variable-object)
+
+### [Server Variable](https://github.com/OAI/OpenAPI-Specification/blob/OpenAPI.next/versions/3.0.md#server-variable-object) object
+
+- [ ] enum
+- [ ] default
+- [ ] description
+
+### [Components](https://github.com/OAI/OpenAPI-Specification/blob/OpenAPI.next/versions/3.0.md#components-object) object
+
+This is supported by default as all `$ref` are dereferenced before the definition is visualised. As per spec, Components have _no impact_ on visualising the API reference, they are simply a placeholder for reusable objects.
+
+- [x] [schemas](#schema-object)
+- [x] [responses](#responses-object)
+- [x] [parameters](#parameter-object)
+- [x] [examples](#example-object)
+- [x] [requestBodies](#request-body-object)
+- [x] [headers](#header-object)
+- [x] [securitySchemes](#security-scheme-object)
+- [x] [links](#link-object)
+- [x] [callbacks](#callback-object)
+
```

Comment:
Yeah, read the little note above that :) Components is meant to be the place where reusable schemas can be defined. Because we dereference the entire file, any `$ref` will be pulled inline, essentially meaning components are required for our purposes.